### PR TITLE
User/rcadene/2024 10 07 vla

### PR DESCRIPTION
**Pull request by danaaubakirova**
## What this does
Explain what this PR does. Feel free to tag your PR with the appropriate label(s).

Examples:
|  Title               | Label           |
|----------------------|-----------------|
| Fixes #[issue]       | (🐛 Bug)        |
| Adds new dataset     | (🗃️ Dataset)    |
| Optimizes something  | (⚡️ Performance) |

## How it was tested
Explain/show how you tested your changes.

Examples:
- Added `test_something` in `tests/test_stuff.py`.
- Added `new_feature` and checked that training converges with policy X on dataset/environment Y.
- Optimized `some_function`, it now runs X times faster than previously.

## How to checkout & try? (for the reviewer)
Provide a simple way for the reviewer to try out your changes.

Examples:
```bash
DATA_DIR=tests/data pytest -sx tests/test_stuff.py::test_something
```
```bash
python lerobot/scripts/train.py --some.option=true
```

## SECTION TO REMOVE BEFORE SUBMITTING YOUR PR
**Note**: Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR. Try to avoid tagging more than 3 people.

**Note**: Before submitting this PR, please read the [contributor guideline](https://github.com/huggingface/lerobot/blob/main/CONTRIBUTING.md#submitting-a-pull-request-pr).

### CODE REVIEW

File: lerobot/common/policies/vla/modeling_vla.py [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+        breakpoint()
+        # Forward pass through Llava (to get hidden states)
+        llava_output = self.language_model(  # Calling the Llava model inside VLA
+            **processed_inputs,
+            return_dict=True,
+            output_hidden_states=True
+        )
+        
+        hidden_states = llava_output.hidden_states[-1]  # Use last layer's hidden state
+        hidden_states = hidden_states[:, -4:, :] #make 4 a config parameter
```
Comment (danaaubakirova) [inline lerobot/common/policies/vla/modeling_vla.py:98]: Taking the last 4 embeddings as an input to the action decoder. Because chunk_size is divisible by 4.

File: lerobot/common/policies/vla/modeling_vla.py [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+        if len(self.expected_image_keys) > 0:
+            batch = dict(batch)  
+            batch["observation.images"] = torch.stack([batch[k] for k in self.expected_image_keys], dim=-4).to(self.device)
+        
+        batch["prompt"] = self.config.prompt.to(self.device)
+        
+        # Process inputs (text and images) using the processor
+        processed_inputs = self.processor(
+            text=batch["prompt"], videos=list(batch["observation.images"]), return_tensors="pt", padding=True, do_rescale=False
+        ).to(self.device)
```
Comment (danaaubakirova) [inline lerobot/common/policies/vla/modeling_vla.py:86]: for some reasons I lose the original batch size? it should be 2. Maybe because we have a 5-dim input, not sure what to do with the camera index. I remove normalization as well, because the processor takes unnormalized PIL images/frames

File: lerobot/common/policies/vla/modeling_vla.py [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+            """
+            Forward pass to compute action logits using hidden states from Qwen2VL (Llava).
+            
+            Args:
+            hidden_states: Tensor of shape [batch_size, seq_len, hidden_size] from Llava model.
+            
+            Returns:
+            action_logits: Tensor of predicted actions.
+            """
+            batch_size = hidden_states.size(0)  # Ensure batch size is extracted
```
Comment (danaaubakirova) [inline lerobot/common/policies/vla/modeling_vla.py:280]: This part is the core and I think it might contain the mistakes. the way we repeat the input to match the chunk_size and also the encoder_out value needs to be checked and reviewed.

File: lerobot/common/policies/vla/modeling_vla.py (hunk 2 above)
Comment (danaaubakirova) [reply lerobot/common/policies/vla/modeling_vla.py:82]: resolved
